app/javascript/src/Common/components/SelectWithModal.jsx

```diff
@@ -1,149 +1,171 @@
 // @flow
 
 import * as React from 'react'
+import { useState, useEffect } from 'react'
 
-import {
-  FormGroup,
-  Select,
-  SelectVariant
-} from '@patternfly/react-core'
-import {
-  toSelectOption,
-  toSelectOptionObject,
-  SelectOptionObject,
-  handleOnFilter
-} from 'utilities'
-import { TableModal } from 'Common'
+import { SortByDirection, sortable } from '@patternfly/react-table'
+import escapeRegExp from 'lodash.escaperegexp'
+import { FancySelect, TableModal } from 'Common'
+import { paginateCollection } from 'utilities'
 
 import type { Record } from 'utilities'
+import type { FetchItemsRequestParams, FetchItemsResponse } from 'Types'
 
 import './SelectWithModal.scss'
 
 type Props<T: Record> = {
+  label: string,
+  fieldId: string,
+  id: string,
+  name: string,
   item: T | null,
-  items: T[],
+  items: Array<T>,
+  itemsCount: number,
+  cells: Array<{ title: string, propName: string, transforms?: [typeof sortable] }>,
   onSelect: (T | null) => void,
+  header: string,
   isDisabled?: boolean,
-  isValid?: boolean,
-  label: string,
-  id: string,
-  name?: string,
-  helperText?: React.Node,
+  title: string,
+  placeholder: string,
+  footerLabel: string,
   helperTextInvalid?: string,
-  placeholderText?: string,
-  maxItems?: number,
-  header?: string,
-  footer?: string,
-  cells: { title: string, propName: string }[],
-  modalTitle: string,
+  fetchItems: (params: FetchItemsRequestParams) => FetchItemsResponse<T>
 }
 
+const PER_PAGE = 5
 const MAX_ITEMS = 20
-const HEADER = 'Most recently created'
-const FOOTER = 'View all'
 
 const SelectWithModal = <T: Record>({
-  item,
-  items,
-  onSelect,
-  isDisabled,
-  isValid,
   label,
+  fieldId,
   id,
   name,
-  helperText,
-  helperTextInvalid,
-  placeholderText,
-  maxItems = MAX_ITEMS,
-  header = HEADER,
-  footer = FOOTER,
+  item,
+  items: initialItems,
```

Comment:
Typo?
Where is `initialItems` defined? is not in the type definition

Reply:
`initialItems` is a label since `items` will be mutatable once `fetchItems` is implemented. This is a middle step.